Sources/NIO/PriorityQueue.swift

```diff
@@ -12,46 +12,56 @@
 //
 //===----------------------------------------------------------------------===//
 
+@usableFromInline
 internal struct PriorityQueue<Element: Comparable> {
-    private var heap: Heap<Element>
+    @usableFromInline
+    internal var _heap: Heap<Element>
 
-    public init(ascending: Bool = false) {
-        self.heap = Heap(type: ascending ? .minHeap : .maxHeap)
+    internal init(ascending: Bool = false) {
+        self._heap = Heap(type: ascending ? .minHeap : .maxHeap)
     }
 
-    public mutating func remove(_ key: Element) {
-        self.heap.remove(value: key)
+    @inlinable
+    internal mutating func remove(_ key: Element) {
+        self._heap.remove(value: key)
     }
 
-    public mutating func push(_ key: Element) {
-        self.heap.append(key)
+    @inlinable
```

Comment:
Similarly, what are the gains of making this function `@inlinable`?

Reply:
In fact isn't this never called from anywhere that is itself `@inlinable`? `SelectableEventLoop.schedule0` is `@usableFromInline`, and `EmbeddedEventLoop`'s `push`/`pop` callers aren't `@inlinable` either, so I'd have assumed this attribute does nothing.

Reply:
I'd leave this one. I just haven't done the work yet for everything in the heap to be inlinable too because right now it doesn't help with anything.

Reply:
I mean, I don't care about leaving it in, but I don't think it'll ever help. The heap is only ever used with one concrete type, defined in NIO so it'll always be specialised, and these call sites are several layers deep in the call stack such that their inability to be inlined is basically irrelevant.

Reply:
@Lukasa yes, only if we'd ever re-expose PriorityQueue or if someone copied the code. I thought if we need to stick an attribute here, `@inlinable` is the better one. But 100% agreed it doesn't help with anything right now.